extensions/infinispan-embedded/deployment/src/main/java/io/quarkus/infinispan/embedded/deployment/InfinispanEmbeddedProcessor.java

```diff
@@ -0,0 +1,247 @@
+            if (!interfaceImplementations.isEmpty()) {
+                serviceProvider
+                        .produce(new ServiceProviderBuildItem(serviceLoadedInterface.getName(), interfaceImplementations));
+            }
+        }
+
+        // These are either default or required for marshalling
+        resources.produce(new SubstrateResourceBuildItem(
+                "org/infinispan/protostream/message-wrapping.proto",
+                "proto/generated/persistence.commons.proto",
+                "proto/generated/persistence.core.proto",
+                "default-configs/default-jgroups-udp.xml",
+                "default-configs/default-jgroups-tcp.xml",
+                "default-configs/default-jgroups-kubernetes.xml",
+                "default-configs/default-jgroups-ec2.xml",
+                "default-configs/default-jgroups-google.xml",
+                "default-configs/default-jgroups-azure.xml"));
+
+        reflectiveClass.produce(new ReflectiveClassBuildItem(false, false, HashFunctionPartitioner.class));
+        reflectiveClass.produce(new ReflectiveClassBuildItem(false, false, JGroupsTransport.class));
+
+        // XML reflection classes
+        reflectiveClass.produce(new ReflectiveClassBuildItem(false, false,
+                "com.sun.org.apache.xerces.internal.jaxp.DocumentBuilderFactoryImpl",
+                "com.sun.org.apache.xerces.internal.jaxp.datatype.DatatypeFactoryImpl",
+                "com.sun.org.apache.xalan.internal.xsltc.trax.TransformerFactoryImpl",
+                "com.sun.org.apache.xerces.internal.jaxp.SAXParserFactoryImpl",
+                "com.sun.xml.bind.v2.ContextFactory",
+                "com.sun.xml.internal.bind.v2.ContextFactory",
+                "com.sun.xml.internal.stream.XMLInputFactoryImpl"));
+
+        CollectionExternalizer.getSupportedPrivateClasses()
+                .forEach(ceClass -> reflectiveClass.produce(new ReflectiveClassBuildItem(false, false, ceClass)));
+        MapExternalizer.getSupportedPrivateClasses()
+                .forEach(ceClass -> reflectiveClass.produce(new ReflectiveClassBuildItem(false, false, ceClass)));
+
+        Set<DotName> excludedClasses = new HashSet<>();
+        excludedReflectionClasses.forEach(excludedBuildItem -> excludedClasses.add(excludedBuildItem.getExcludedClass()));
+
+        IndexView combinedIndex = combinedIndexBuildItem.getIndex();
+
+        // Add all the JGroups Protocols
+        addReflectionForName(Protocol.class.getName(), false, combinedIndex, reflectiveClass, false, true, excludedClasses);
+
+        // Add all the JGroups Property Converters
+        addReflectionForClass(PropertyConverter.class, true, combinedIndex, reflectiveClass, excludedClasses);
+
+        // Just add the JBossAS and Embedded TransactionManagerLookups
+        reflectiveClass.produce(new ReflectiveClassBuildItem(false, false, JBossStandaloneJTAManagerLookup.class));
```

Comment:
The `JBossStandaloneJTAManagerLookup` is so the user can link Infinispan with the transaction manager in Quarkus. The `EmbeddedTransactionManagerLookup` is if they just want a simple transaction manager to use outside of it.

> Also, I think you should auto-inject the right choice - right now I see the documentation require the user to set one, this should be unnecessary if you can require the single choice of the platform.

Do you mean to say we should only allow the Quarkus transaction manager? Then the user wouldn't have to configure that. This would require the transaction extension always then. The embedded one is quite light weight in comparison.

Reply:
Yes support only "the" transaction manager provided by the platform.

Many optimisations are made possible only because we give up on allowing people to use any weird and crazy replacement for core components: it also helps quality & people as we're accountable for the choice of implementations being made.

Reply:
👍 that makes my life easier as well